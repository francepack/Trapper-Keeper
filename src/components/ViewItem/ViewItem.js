import React from 'react'
import PropTypes from 'prop-types'

export const ViewItem = (props) => {
  let style
  if(props.complete === 'true') {
    style = 'list-item strikethrough view-item'
  } else {
    style = 'list-item view-item'
  }
  return(
    <div className='list-container view-item'>
      <textarea defaultValue={props.existingValue}
                className={style}
              >
      </textarea>
    </div>
  )
}

ViewItem.propTypes = {
  existingValue: PropTypes.string
}

export default ViewItem