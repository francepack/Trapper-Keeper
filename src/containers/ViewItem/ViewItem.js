import React from 'react'

export const ViewItem = (props) => {
  return(
    <div className='list-container'>
      <button className='list-control'></button>
      <textarea defaultValue={props.existingValue}
                className='list-item'
                >
      </textarea>
    </div>
  )
}

export default ViewItem