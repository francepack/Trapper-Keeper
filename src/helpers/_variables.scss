
$background: #fff;
$logo: #f4b70b;
$placeholder: #767676;
$type: #202124;
$icon: #606163;
$hover: #e8eaed;
$form-border: #e0e0e0;
$save-btn: #4285F4;
$save-btn-hover: #4285f4d6;
$check-mark: #0F9D58;
$delete-btn: #DB4437;
$delete-hover: #db4437c9;