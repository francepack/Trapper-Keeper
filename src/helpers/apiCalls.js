export const addNote = async (note) => {
  const url = 'http://localhost:3001/api/v1/notes'
  try {
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify(note),
      headers: {
        'Content-Type': 'application/json'
      },
    })
    const addedNote = await response.json()
    return addedNote;
  } catch (error) {
    return error.message
  }
}

export const getNotes = async () => {
  const url = 'http://localhost:3001/api/v1/notes'
  try {
    const response = await fetch(url);
    const allNotes = await response.json();
    return allNotes.notes
  } catch (error) {
    return error.message
  }
}

export const removeNote = async (id) => {
  const url = `http://localhost:3001/api/v1/notes/${id}`
  try {
    const response = await fetch(url, {
      method: 'DELETE',
      body: JSON.stringify(id),
      headers: {
        'Content-Type': 'application/json'
      },
    })

  } catch(error) {
    return error.message
  }
}