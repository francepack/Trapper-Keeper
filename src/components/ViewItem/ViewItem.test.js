import React from 'react'
import { ViewItem } from './ViewItem'
import { shallow } from 'enzyme'
import { mockNote } from '../../helpers/mockData'

describe('ViewItem', () => {
  let wrapper
  it('should match snapshot', () => {
    wrapper = shallow(
      <ViewItem {...mockNote.items[0].value}/>
    )

    expect(wrapper).toMatchSnapshot()
  })
})