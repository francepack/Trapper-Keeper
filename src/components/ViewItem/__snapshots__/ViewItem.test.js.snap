// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewItem should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewItem
    0="f"
    1="i"
    2="r"
    3="s"
    4="t"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <textarea
        className="list-item view-item"
      />,
      "className": "list-container view-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "list-item view-item",
        "defaultValue": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <textarea
          className="list-item view-item"
        />,
        "className": "list-container view-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "list-item view-item",
          "defaultValue": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
